src/CommonSocket.php

```diff
@@ -240,8 +262,149 @@ protected function createStream()
                     . ' (' . $errstr . ' [' . $errno . '])';
             throw new \Kafka\Exception($error);
         }
+        // SASL auth
+        if ($this->provider != null) {
+            $this->provider->authenticate($this);
+        }
     }
 
+    // }}}
+    // {{{ public function selectRead()
+
+    /**
+     * Read from the socket at most $len bytes.
+     *
+     * This method will not wait for all the requested data, it will return as
+     * soon as any data is received.
+     *
+     * @param integer $len               Maximum number of bytes to read.
+     * @param boolean $verifyExactLength Throw an exception if the number of read bytes is less than $len
+     *
+     * @return string Binary data
+     * @throws \Kafka\Exception
+     */
+    public function selectRead($len, $verifyExactLength = false)
+    {
+        if ($len > self::READ_MAX_LEN) {
+            throw new \Kafka\Exception('Could not read ' . $len . ' bytes from stream, length too longer.');
```

Comment:
We didn't even try to read so `Could not read` is a bit misleading, maybe:

```php
throw new \Kafka\Exception('Invalid length given, it should be lesser than or equals to ' . self:: READ_MAX_LEN);
```

Reply:
It would also be amazing to rename `self::READ_MAX_LEN` to `self:: READ_MAX_LENGTH`